### ISSUE TITLE
primer_search does not seem to parse Primer-Blast results correctly

### ISSUE THREAD

**Issue by holgerbrandl** [MVesuviusC/primerTree#18]
According to the docs `primer_search` should return a data.frame with primer hits but using the following example it does not seem to work:

``` r
require_auto(primerTree)
forwardPrimer="GATGCATCACCATGAACCGC"
reversePrimer="GGTTTGAACATCGCCGATCG"

result <- primer_search(forwardPrimer, reversePrimer)
> BLASTing 1 primer combinations
> No encoding supplied: defaulting to UTF-8.
> Submitting Primer-BLAST query
> No encoding supplied: defaulting to UTF-8.
> BLAST alignment processing, refreshing in 20 seconds...
> No encoding supplied: defaulting to UTF-8.
> BLAST alignment processing, refreshing in 20 seconds...
> No encoding supplied: defaulting to UTF-8.
> BLAST alignment processing, refreshing in 20 seconds...
> No encoding supplied: defaulting to UTF-8.
> BLAST alignment completed in 124 seconds
> There were 25 warnings (use warnings() to see them)
is.data.frame(result)
> FALSE
```

There are many warning which seem to indicate that the query failed:

```
> warnings()
Warning messages:
1: In GET_retry(url) : request failed, retry attempt 1
2: In GET_retry(url) : request failed, retry attempt 2
3: In GET_retry(url) : request failed, retry attempt 3
4: In GET_retry(url) : request failed, retry attempt 4
5: In GET_retry(url) : request failed, retry attempt 5
6: In POST_retry(paste(url, "primertool.cgi", sep = ""),  ... :
  request failed, retry attempt 1
7: In POST_retry(paste(url, "primertool.cgi", sep = ""),  ... :
  request failed, retry attempt 2
8: In POST_retry(paste(url, "primertool.cgi", sep = ""),  ... :
  request failed, retry attempt 3
9: In POST_retry(paste(url, "primertool.cgi", sep = ""),  ... :
  request failed, retry attempt 4
10: In POST_retry(paste(url, "primertool.cgi", sep = ""),  ... :
  request failed, retry attempt 5
11: In GET_retry(values[2]) : request failed, retry attempt 1
12: In GET_retry(values[2]) : request failed, retry attempt 2
13: In GET_retry(values[2]) : request failed, retry attempt 3
14: In GET_retry(values[2]) : request failed, retry attempt 4
15: In GET_retry(values[2]) : request failed, retry attempt 5
16: In GET_retry(values[2]) : request failed, retry attempt 1
17: In GET_retry(values[2]) : request failed, retry attempt 2
18: In GET_retry(values[2]) : request failed, retry attempt 3
19: In GET_retry(values[2]) : request failed, retry attempt 4
20: In GET_retry(values[2]) : request failed, retry attempt 5
21: In GET_retry(values[2]) : request failed, retry attempt 1
22: In GET_retry(values[2]) : request failed, retry attempt 2
23: In GET_retry(values[2]) : request failed, retry attempt 3
24: In GET_retry(values[2]) : request failed, retry attempt 4
25: In GET_retry(values[2]) : request failed, retry attempt 5
```

Not sure if it's helpful but the structure of the `result` is as follows:

```
List of 1
 $ 1:List of 10
  ..$ url        : chr "http://www.ncbi.nlm.nih.gov/tools/primer-blast/primertool.cgi?ctg_time=1455194104&job_key=KTLJJeAvSQt2OXI3ExlASQg0clgbK29d"
  ..$ status_code: int 200
  ..$ headers    :List of 15
  .. ..$ date                            : chr "Thu, 11 Feb 2016 12:36:16 GMT"
  .. ..$ server                          : chr "Apache"
  .. ..$ cache-control                   : chr "private"
  .. ..$ ncbi-phid                       : chr "9CAA62436BC7FF810000000000001375"
  .. ..$ ncbi-sid                        : chr "075137C36BC6CA31_0000SID"
  .. ..$ access-control-allow-methods    : chr "POST, GET, PUT, OPTIONS, PATCH, DELETE"
  .. ..$ access-control-allow-origin     : chr ""
  .. ..$ access-control-allow-credentials: chr "true"
  .. ..$ access-control-allow-headers    : chr "X-Accept-Charset,X-Accept,Content-Type,X-Requested-With,NCBI-SID,NCBI-PHID"
  .. ..$ content-type                    : chr "text/html"
  .. ..$ set-cookie                      : chr "ncbi_sid=075137C36BC6CA31_0000SID; domain=.nih.gov; path=/; expires=Sat, 11 Feb 2017 12:35:04 GMT"
  .. ..$ vary                            : chr "Accept-Encoding"
  .. ..$ content-encoding                : chr "gzip"
  .. ..$ x-ua-compatible                 : chr "IE=Edge"
  .. ..$ transfer-encoding               : chr "chunked"
  .. ..- attr(*, "class")= chr [1:2] "insensitive" "list"
  ..$ all_headers:List of 1
  .. ..$ :List of 3
  .. .. ..$ status : int 200
  .. .. ..$ version: chr "HTTP/1.1"
  .. .. ..$ headers:List of 15
  .. .. .. ..$ date                            : chr "Thu, 11 Feb 2016 12:36:16 GMT"
  .. .. .. ..$ server                          : chr "Apache"
  .. .. .. ..$ cache-control                   : chr "private"
  .. .. .. ..$ ncbi-phid                       : chr "9CAA62436BC7FF810000000000001375"
  .. .. .. ..$ ncbi-sid                        : chr "075137C36BC6CA31_0000SID"
  .. .. .. ..$ access-control-allow-methods    : chr "POST, GET, PUT, OPTIONS, PATCH, DELETE"
  .. .. .. ..$ access-control-allow-origin     : chr ""
  .. .. .. ..$ access-control-allow-credentials: chr "true"
  .. .. .. ..$ access-control-allow-headers    : chr "X-Accept-Charset,X-Accept,Content-Type,X-Requested-With,NCBI-SID,NCBI-PHID"
  .. .. .. ..$ content-type                    : chr "text/html"
  .. .. .. ..$ set-cookie                      : chr "ncbi_sid=075137C36BC6CA31_0000SID; domain=.nih.gov; path=/; expires=Sat, 11 Feb 2017 12:35:04 GMT"
  .. .. .. ..$ vary                            : chr "Accept-Encoding"
  .. .. .. ..$ content-encoding                : chr "gzip"
  .. .. .. ..$ x-ua-compatible                 : chr "IE=Edge"
  .. .. .. ..$ transfer-encoding               : chr "chunked"
  .. .. .. ..- attr(*, "class")= chr [1:2] "insensitive" "list"
  ..$ cookies    :'data.frame': 1 obs. of  7 variables:
  .. ..$ domain    : chr ".nih.gov"
  .. ..$ flag      : logi TRUE
  .. ..$ path      : chr "/"
  .. ..$ secure    : logi FALSE
  .. ..$ expiration: POSIXct[1:1], format: "2017-02-11 13:35:04"
  .. ..$ name      : chr "ncbi_sid"
  .. ..$ value     : chr "075137C36BC6CA31_0000SID"
  ..$ content    : raw [1:6629] ef bb bf 3c ...
  ..$ date       : POSIXct[1:1], format: "2016-02-11 12:36:16"
  ..$ times      : Named num [1:6] 0 0.000027 0.000029 0.000075 0.122304 ...
  .. ..- attr(*, "names")= chr [1:6] "redirect" "namelookup" "connect" "pretransfer" ...
  ..$ request    :List of 7
  .. ..$ method    : chr "GET"
  .. ..$ url       : chr "http://www.ncbi.nlm.nih.gov/tools/primer-blast/primertool.cgi?ctg_time=1455194104&job_key=KTLJJeAvSQt2OXI3ExlASQg0clgbK29d"
  .. ..$ headers   : Named chr "application/json, text/xml, application/xml, */*"
  .. .. ..- attr(*, "names")= chr "Accept"
  .. ..$ fields    : NULL
  .. ..$ options   :List of 2
  .. .. ..$ useragent    : chr "libcurl/7.43.0 r-curl/0.9.5 httr/1.1.0"
  .. .. ..$ customrequest: chr "GET"
  .. ..$ auth_token: NULL
  .. ..$ output    : list()
  .. .. ..- attr(*, "class")= chr [1:2] "write_memory" "write_function"
  .. ..- attr(*, "class")= chr "request"
  ..$ handle     :Class 'curl_handle' <externalptr> 
  ..- attr(*, "class")= chr "response"
 - attr(*, "split_type")= chr "array"
 - attr(*, "split_labels")='data.frame':    1 obs. of  1 variable:
  ..$ X1: int 1
```

I'm using the latest version of primerTree from github and here's my sessionInfo:

```
> sessionInfo()
R version 3.2.3 (2015-12-10)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X 10.10.5 (Yosemite)

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] primerTree_1.0.3        gridExtra_2.0.0         directlabels_2015.12.16 scales_0.3.0            ggplot2_2.0.0           readxl_0.1.0.9000      
 [7] readr_0.2.2             digest_0.6.9            tidyr_0.4.0             magrittr_1.5            dplyr_0.4.3             reshape2_1.4.1         
[13] stringr_1.0.0           plyr_1.8.3             

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.3      devtools_1.10.0  munsell_0.4.2    lattice_0.20-33  ape_3.4          colorspace_1.2-6 R6_2.1.2         quadprog_1.5-5  
 [9] foreach_1.4.3    httr_1.1.0       tools_3.2.3      parallel_3.2.3   nlme_3.1-124     gtable_0.1.2     DBI_0.3.1        iterators_1.0.8 
[17] assertthat_0.1   codetools_0.2-14 curl_0.9.5       memoise_0.2.1    stringi_1.0-1    XML_3.98-1.3     lubridate_1.5.0 
```

**Comment by jimhester** [comment]
So there were two issues with this. First a capitalization change in httr broke the retry test (fixed 8143ec8362).

Second the primers you are searching produce no results, if you open the url from your response (http://www.ncbi.nlm.nih.gov/tools/primer-blast/primertool.cgi?ctg_time=1455194104&job_key=KTLJJeAvSQt2OXI3ExlASQg0clgbK29d) you will see

> No target templates were found in selected database: Nucleotide collection (nt).

This case should be handled better by primerTree, but unfortunately I don't have the bandwidth to do so currently.

**Comment by holgerbrandl** [comment]
Thanks for the patch. However, the problem is actually a different one. `primer_search` does simply not do what it is supposed to according to the docs, which is `@return data.frame of primer hits`. It returns the html response object from NCBI-blast instead. To get the actual data.frame of primer hits the user has to 

``` r
 primer_search(forwardPrimer, reversePrimer) %>% parse_primer_hits()
```

This works for me. So I think it's just the docs (and maybe the example in the README.md) that need to be improved.
